runtime/drivers/clickhouse/clickhouse.go

```diff
@@ -507,39 +507,70 @@ func (c *Connection) lastUsedOn() time.Time {
 }
 
 // Periodically collects stats about the database and emit them as activity events.
-func (c *Connection) periodicallyEmitStats(d time.Duration) {
+func (c *Connection) periodicallyEmitStats(sensitive, regular time.Duration) {
 	if c.activity == nil {
 		// Activity client isn't set, there is no need to report stats
 		return
 	}
 
-	ticker := time.NewTicker(d)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-ticker.C:
-			// Skip if it hasn't been used recently and may be scaled to zero.
-			if c.config.CanScaleToZero && time.Since(c.lastUsedOn()) > 2*d {
-				continue
+	// Sensitive ticker for sensitive stats
+	sensitiveTicker := time.NewTicker(sensitive)
+	defer sensitiveTicker.Stop()
+
+	// Regular ticker for non-sensitive stats
+	regularTicker := time.NewTicker(regular)
+	defer regularTicker.Stop()
+
+	// Emit non-sensitive stats periodically
+	go func() {
+		for {
+			select {
+			case <-regularTicker.C:
+				// Emit the latest RCU per service.
+				latestRCU, err := c.latestRCUPerService(c.ctx)
+				if err == nil {
+					for service, value := range latestRCU {
+						c.activity.RecordMetric(c.ctx, "clickhouse_rcu", value, attribute.String("billing_service", service))
+					}
+					if len(latestRCU) == 0 {
+						c.logger.Warn("no RCU data found for any service", zap.String("clickhouse_host", c.config.Host))
+					}
+				} else if !errors.Is(err, c.ctx.Err()) {
+					c.logger.Warn("failed to fetch latest RCU per service", zap.Error(err))
+				}
+			case <-c.ctx.Done():
+				return
 			}
-
-			// Emit the estimated size of the database.
-			size, err := c.estimateSize(c.ctx)
-			if err == nil {
-				c.activity.RecordMetric(c.ctx, "clickhouse_estimated_size_bytes", float64(size))
-			} else if !errors.Is(err, c.ctx.Err()) {
-				lvl := zap.WarnLevel
-				if c.config.Managed {
-					lvl = zap.ErrorLevel
+		}
+	}()
+
+	// Emit sensitive stats periodically
+	go func() {
```

Comment:
It can use the same loop and just have multiple `case` statements in the `select` for the different tickers. The call to `periodicallyEmitStats` is already running in the background (invoked with `go`), so ideally it should avoid internally creating more background tasks